configure.ac

```
@@ -4081,14 +4081,20 @@ PY_CHECK_FUNC([setgroups], [
 #endif
 ])
 
-# check for openpty and forkpty
+# check for openpty, login_tty, and forkpty
 
 AC_CHECK_FUNCS(openpty,,
    AC_CHECK_LIB(util,openpty,
      [AC_DEFINE(HAVE_OPENPTY) LIBS="$LIBS -lutil"],
      AC_CHECK_LIB(bsd,openpty, [AC_DEFINE(HAVE_OPENPTY) LIBS="$LIBS -lbsd"])
    )
 )
+AC_CHECK_FUNCS(login_tty,,
+   AC_CHECK_LIB(util,login_tty,
+     [AC_DEFINE(HAVE_LOGIN_TTY) LIBS="$LIBS -lutil"],
+     AC_CHECK_LIB(bsd,login_tty, [AC_DEFINE(HAVE_LOGIN_TTY) LIBS="$LIBS -lbsd"])
+   )
+)
```

This approach is easier and better. It updates ``LIBS`` only when it is necessary and won't add ``libutils`` a second time. The ``login_tty`` is provided by libutil on Linux, FreeBSD, OpenBSD, and NetBSD.

```suggestion
AC_SEARCH_LIBS([login_tty], [util],
 [AC_DEFINE([HAVE_LOGIN_TTY], [1], [Define to 1 if you have the `login_tty' function.])]
)
```